spikeextractors/extractors/cellexplorersortingextractor/cellexplorersortingextractor.py

```diff
@@ -0,0 +1,102 @@
+from spikeextractors import SortingExtractor
+import numpy as np
+from pathlib import Path
+from spikeextractors.extraction_tools import check_valid_unit_id
+from typing import Union
+from scipy.io import loadmat, savemat
+
+PathType = Union[str, Path, None]
+
+
+class CellExplorerSortingExtractor(SortingExtractor):
+    """
+    Extracts spiking information from .mat files stored in the CellExplorer format.
+
+    Spike times are stored in units of seconds.
+
+    Parameters
+    ----------
+    spikes_matfile_path : PathType
+        Path to the sorting_id.spikes.cellinfo.mat file.
+    """
+
+    extractor_name = "CellExplorerSortingExtractor"
+    installed = True
+    is_writable = True
+    mode = "file"
+    installation_mesg = ""
+
+    def __init__(self, spikes_matfile_path: PathType):
```

Comment:
It seems like since you are reading data from more than one file in this directory you should give the directory path instead of the filepath.

Reply:
While it would totally be doable to swap it to a `folder_path` setup instead of `file_path`, I'm always basing these designs and styles after other extractor types. SpikeGLX and Neuroscope both have this same type of structure where you're reading the actual data from the primary `file_path`, but also pulling metadata from a secondary file in the same folder.

@alejoe91 Which mode do you think would be more appropriate in these types of situations?

Reply:
I see... I'll defer to @alejoe91 on that then